### ISSUE TITLE
Add the option to sorts map by date

### ISSUE THREAD

**Issue by jnp9** [doug-m/CollectorAuroraTesting#120]
### Issue

Provide the option to sort your maps by date. Similar to what is offered in Survey123. Currently Collector only sorts by alphabetical order.

### Expected Behavior

Use case: If you have a large number of maps available it can be cumbersome to scroll through them to get to your map. Being able to sort by date so you see the most recent maps first would be very handy. This functionality exists in Survey123.

### Map and Credentials

N\A

### Steps to Reproduce

Open Collector and access your maps.

### Device Information
* Device: iPhone 5
* OS Version: 11

N\A

**Comment by mudnmaps** [comment]
I agree that would be helpful as we have a dozen maps deployed for some users.

**Comment by Kevin-GIS** [comment]
In SF 873, a Groups section was exposed in the maps browse screen. Within each group is the ability to sort by date and title. 

Those options are not available outside of the groups though.
